framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -191,6 +191,76 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
                 " is consistently oriented with regards to the blocks it neighbors";
 
     diagnosticsLog(message, _check_sidesets_orientation, flipped_pairs.size());
+
+    // Now check that there is no sideset radically flipping from one side's normal to another
+    // side next to it, in the same sideset
+    // We'll consider pi / 2 to be the most steep angle we'll pass
+    unsigned int num_normals_flipping = 0;
+    Real steepest_side_angles = 1;
+    for (const auto & [elem_id, side_id, side_bid] : side_tuples)
+    {
+      if (side_bid != bid)
+        continue;
+      const auto & elem_ptr = mesh->elem_ptr(elem_id);
+
+      // Get side normal
+      const std::unique_ptr<const Elem> face = elem_ptr->build_side_ptr(side_id);
+      std::unique_ptr<FEBase> fe(FEBase::build(elem_ptr->dim(), FEType(elem_ptr->default_order())));
+      QGauss qface(elem_ptr->dim() - 1, CONSTANT);
+      fe->attach_quadrature_rule(&qface);
+      const auto & normals = fe->get_normals();
+      fe->reinit(elem_ptr, side_id);
+      mooseAssert(normals.size() == 1, "We expected only one normal here");
+      const auto & side_normal = normals[0];
+
+      // Compare to the sideset normals of neighbor sides in that sideset
+      for (const auto neighbor : elem_ptr->neighbor_ptr_range())
+        if (neighbor)
+          for (const auto neigh_side_index : neighbor->side_index_range())
+          {
+            // Check that the neighbor side is also in the sideset being examined
+            if (!boundary_info.has_boundary_id(neighbor, neigh_side_index, bid))
+              continue;
+
+            // We re-init everything for the neighbor in case it's a different dimension
```

Comment:
I don't believe it's possible for a neighbor to be of a different dimension. @roystgnr ?

Reply:
Nope; a neighbor should always be of the same dimension.  A `reinit()` is still necessary in general, but the `build()` and accouterments aren't; the FE object can get pulled outside the loop.

Reply:
I ll trim that up on the next diagnostics. There's still plenty on the llist